common/lib/dependabot/pull_request_creator/github.rb

```diff
@@ -347,6 +350,18 @@ def add_milestone_to_pull_request(pull_request)
       end
 
       def create_pull_request
+        # Limit PR description to MAX_PR_DESCRIPTION_LENGTH (65,536) characters
+        # and truncate with message if over. The API limit is 262,144 bytes
+        # (https://github.community/t/maximum-length-for-the-comment-body-in-issues-and-pr/148867/2).
+        # As Ruby strings are UTF-8 encoded, this is a pessimistic limit: it
+        # presumes the case where all characters are 4 bytes.
+        pr_description = @pr_description.dup
+        if pr_description && pr_description.length > MAX_PR_DESCRIPTION_LENGTH
```

Comment:
This is similar to a block in `common/lib/dependabot/clients/azure.rb`, which also has to deal with encoding issues. Chose to copy in preference of a [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) refactor.

Reply:
Thank you